bom/pom.xml

```diff
@@ -0,0 +1,374 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <name>grpc-bom</name>
+  <description>gRPC BOM for Java</description>
+  <groupId>io.grpc</groupId>
+  <artifactId>grpc-bom</artifactId>
+  <version>1.19.0-SNAPSHOT</version><!-- CURRENT_GRPC_VERSION -->
+  <packaging>pom</packaging>
+  <url>https://github.com/grpc/grpc-java</url>
+
+  <licenses>
+    <license>
+      <name>Apache License 2.0</name>
+      <url>https://github.com/grpc/grpc-java/blob/master/LICENSE</url>
+    </license>
+  </licenses>
+
+  <issueManagement>
+    <system>GitHub</system>
+    <url>https://github.com/grpc/grpc-java/issues</url>
+  </issueManagement>
+
+  <ciManagement>
+    <system>Travis CI</system>
+    <url>https://travis-ci.org/grpc/grpc-java</url>
+  </ciManagement>
+
+  <developers>
+    <developer>
+      <id>grpc.io</id>
+      <name>gRPC Contributors</name>
+      <email>grpc-io@googlegroups.com</email>
+      <url>https://grpc.io/</url>
+      <organization>gRPC Authors</organization>
+      <organizationUrl>https://www.google.com</organizationUrl>
+    </developer>
+  </developers>
+
+  <scm>
+    <connection>scm:git:https://github.com/grpc/grpc-java.git</connection>
+    <url>https://github.com/grpc/grpc-java</url>
+  </scm>
+
+  <properties>
+    <netty.version>4.1.32.Final</netty.version>
+    <netty.tcnative.version>2.0.7.Final</netty.tcnative.version>
+    <guava.version>26.0-android</guava.version>
+    <protobuf.version>3.5.1</protobuf.version>
+    <protoc.version>3.5.1-1</protoc.version>
+    <protoc.nano.version>3.0.0-alpha-5</protoc.nano.version>
+    <opencensus.version>0.18.0</opencensus.version>
+    <googleauth.version>0.9.0</googleauth.version>
+  </properties>
+
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>animal-sniffer-annotations</artifactId>
+        <version>1.17</version>
+      </dependency>
+      <dependency>
+        <groupId>com.google.errorprone</groupId>
+        <artifactId>error_prone_annotations</artifactId>
+        <version>2.2.0</version>
+      </dependency>
+      <dependency>
+        <groupId>com.google.code.gson</groupId>
+        <artifactId>gson</artifactId>
+        <version>2.7</version>
+      </dependency>
+      <dependency>
+        <groupId>com.google.guava</groupId>
+        <artifactId>guava</artifactId>
+        <version>${guava.version}</version>
```

Comment:
Please do not pin versions of transitive dependencies, this will make the BOM useless because it will enforce everyone to use `andoid` and not `jre` version of Guava, Protobuf 3.5.x instead of 3.6.x and so on.

Please see Netty and Jackson as examples of well-formed BOMs:

https://github.com/netty/netty/blob/4.1/bom/pom.xml

https://github.com/FasterXML/jackson-bom/blob/master/pom.xml

Reply:
I agree we should only include `io.grpc` artifacts. For netty the general solution is to use grpc-netty-shaded. If someone specifies grpc-netty directly they will need to manage their netty version.

Reply:
+1 to not including transitive dependencies - for the purposes of cloud-oss-bom (https://github.com/GoogleCloudPlatform/cloud-opensource-java/tree/master/boms/cloud-oss-bom), which combines multiple BOMs together, we need each BOM to be limited to the artifacts of a single library.

Reply:
Repeating myself that only io.grpc should be included (so that it will be part of the review).